core-v1/src/main/scala/data/Justified.scala

```diff
@@ -135,6 +155,15 @@ object Justified {
     }
   }
 
+  implicit val semigroupal: Semigroupal[Justified] = new Semigroupal[Justified] {
+    override def product[A, B](
+      fa: Justified[A],
+      fb: Justified[B],
+    ): Justified[(A, B)] = {
+      Justified.byInference("product", (fa.value, fb.value), NonEmptyList.of(fa, fb))
+    }
+  }
```

Comment:
I want to be able to produce a more flat structure for this. Maybe a special `Justified` type that I can match on to flatten as I build it?